casanovo/config.py

```
@@ -27,6 +27,53 @@
+    TypeError
+        If the value cannot be interpreted as a valid interval.
+    ValueError
+        If a float is outside (0, 1] or an int is less than 1.
+    """
+    try:
+        as_float = float(value)
+    except (TypeError, ValueError) as err:
+        raise TypeError(
+            f"val_check_interval must be an int >= 1 or a float in (0, 1], "
+            f"got {value!r}"
+        ) from err
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)